src/Bicep.Core/CoreResources.resx

```diff
@@ -526,4 +526,14 @@
   <data name="UseSafeAccessRule_MessageFormat" xml:space="preserve">
     <value>The syntax can be simplified by using the safe access (.?) operator.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="WhatIfShortCircuitingRuleDescription" xml:space="preserve">
+    <value>Runtime values cannot be used to determine resource IDs</value>
+  </data>
+  <data name="WhatIfShortCircuitingRuleMessageFormat" xml:space="preserve">
+    <value>Runtime value '{0}' will disable what-if analysis for module '{1}'</value>

```

Comment:
The runtime value won't disable what-if analysis for the whole module, but it will reduce its accuracy. How that will manifest depends a bit on how the runtime value is used: if it's used in a resource condition, what-if will report that the resource may or may not be updated, and it's in used in one of a resource's identifying properties, what-if won't be able to compare what is deployed against a current state.

Maybe something like `Runtime value '{0}' will negatively impact what-if analysis for module '{1}'` or `Runtime value '{0}' will reduce the precision of what-if analysis for module '{1}'`? This may require a bit of wordsmithing to get right.